### ISSUE TITLE
Collaborate with GParted?

### ISSUE THREAD

**Issue by inferno986return** [relan/exfat#106]
As this is a libre implementation of exFAT (well done BTW), I was wondering if there's any way you could collaborate with [GParted](https://en.wikipedia.org/wiki/GParted) (and other partition editing tools) to allow full support for the creation and modification of exFAT partitions.

![screenshot_2018-10-13_18-45-53](https://user-images.githubusercontent.com/1838420/46908261-68d8bf00-cf18-11e8-9d79-d0e8c18514af.png)

In addition, it might also be a good idea to see if the ReactOS team are interested too.

**Comment by relan** [comment]
I'm all for collaboration, but realistically I will hardly have any time for GParted or ReactOS integration. Feel free to drive those features.

**Comment by wdlkmpx** [comment]
Gparted just calls command line apps.. adding support for exfat in GParted is not the problem, the problem is the maintainers.. last time i read something about it i got the they don't want exfat support in gparted.. i guess a few have implemented it privately... maybe @pali has

**Comment by pali** [comment]
Nope, I have not implemented it. I have no motivation to use such rubbish like exfat which is moreover step backward from FAT32. And due to patents problems I doubt that somebody will include exfat implementation into Linux kernel or GParted. Nobody is willing to have problems with M$ law dep.

**Comment by wdlkmpx** [comment]
Well, the problem is not implementing exfat support in gparted, the problem is the maintainers..

**Comment by pali** [comment]
Problem with maintainers? Why? I fully understand maintainers that do not want to include exfat support due to above patent problems. As said, they really do not want to have such problems.

**Comment by wdlkmpx** [comment]
Yeah i understand the patent issues, but people just want to use
exfat.. it's as lightweight as fat32 but better (support 4gb+ files,
and less prone to fs errors and fragmenting?) and some cameras and
hardware only support exfat or something.. in order to please people
you have to break the law..

that's what i read..

**Comment by pali** [comment]
Look, you cannot use something which is disallowed by law. And you said "problem with maintainers" just because they decided not to break low... Nope, this is not a problem. This is correct behavior what maintainers should do.

**Comment by wdlkmpx** [comment]
Relax, life is too short in the GPL too restrictive and viral...

**Comment by inferno986return** [comment]
From my understanding [Microsoft are more lenient with patents with their Open Invention Network](https://www.zdnet.com/article/microsoft-open-sources-its-entire-patent-portfolio/), I did ask on Twitter if that included the Ribbon UI patents and got a like from the representative who tweeted that. Even the FSF seem [cautiously optimistic of the move](https://www.fsf.org/news/fsf-statement-on-microsoft-joining-the-open-invention-network).

However pali does have a point about exFAT having a stigma associated with it and a simple Google Search suggests that due to the filesystem's lack of journalling (and perhaps other factors) there's a heightened risk of data loss.

Still I do dream of a day when GParted's checklist is fully green.

**Comment by inferno986return** [comment]
As a minor update, exFAT is available as a kernel module and Microsoft have pledged to allow the filesystem into the Linux kernel itself according to [OMG! Ubuntu!](https://www.omgubuntu.co.uk/2019/09/install-samsung-exfat-driver-ubuntu-ppa).

Native support would certainly help, would it not?
